<launch>
    <!-- Namespace for the drone -->
    <arg name="namespace" default="iris"/>
    <arg name="world_name" default="basic"/>
    <arg name="paused" default="false"/>
    <arg name="gui" default="true"/>
    <arg name="debug" default="false"/>
    <arg name="verbose" default="false"/>

    <!-- Option to use Gazebo ModelStates (perfect state) instead of Odometry -->
    <arg name="use_model_states_for_control" default="false"/>

    <!-- Launch Gazebo with the Iris model -->
    <include file="$(find fly_bot)/launch/spawn_rotors_iris.launch">
        <arg name="namespace" value="$(arg namespace)"/>
        <arg name="world_name" value="$(arg world_name)"/>
        <arg name="paused" value="$(arg paused)"/>
        <arg name="gui" value="$(arg gui)"/>
        <arg name="debug" value="$(arg debug)"/>
        <arg name="verbose" value="$(arg verbose)"/>
    </include>

    <!-- Launch the CLF Controller Node -->
    <node name="clf_controller" pkg="fly_bot" type="clf_controller_node.py" output="screen">
        <!-- Pass namespace and params -->
        <param name="namespace" value="$(arg namespace)" />
        <param name="use_model_states" value="$(arg use_model_states_for_control)" />

        <!-- Control Targets -->
        <param name="target_altitude" value="3" /> <!-- Target hover altitude -->
        <param name="target_yaw_deg" value="0.0" />  <!-- Target yaw angle -->

        <!-- Yaw Gains (Altitude/Roll/Pitch gains are implicit in CLF) -->
        <param name="kp_yaw" value="2.5" />
        <param name="kd_yaw" value="1.2" />

        <!-- Optional: Override dynamics/motor params if needed -->
        <!-- e.g., <param name="mass" value="1.51" /> -->

        <!-- Control Rate -->
        <param name="control_rate" value="100.0"/> <!-- Hz -->

    </node>

    <!-- Optional: Start rqt_plot or other debugging tools -->
    <!--
    <node name="rqt_plot_z" pkg="rqt_plot" type="rqt_plot"
          args="/clf_controller/error/z" />
    <node name="rqt_plot_u" pkg="rqt_plot" type="rqt_plot"
          args="/clf_controller/control/U1 /clf_controller/control/U2 /clf_controller/control/U3 /clf_controller/control/U4" />
    -->

</launch>
